### ISSUE TITLE
IE8 raises `Expected identifier` exception on `browser-polyfill.js`

### ISSUE THREAD

**Issue by pkese** [babel/babel#1286]
`babel-core/browser-polyfill.js` on a fresh npm install contains `iterator.throw` and `iterator.return` (rather than `["throw"]` and `["return"]`) which consequently makes it incompatible with older browsers.

The actual error was fixed in regenerator a few days ago, but the npm version of babel still installs the old syntax
https://github.com/facebook/regenerator/commit/97466d3d540243702496b24426765f0022f4bc07

To reproduce, I just freshly npm installed babel-core and searched for `.throw` in resulting browser-polyfill.js

**Comment by sebmck** [comment]
It's completely unrelated to the regenerator issue. The problem is the uglify settings that are used to minify `browser-polyfill`.

**Comment by zloirock** [comment]
@sebmck related. IIRC, uglify not transform `iterator["throw"]` -> `iterator.throw`, looks like simply need rebuild `browser-polyfill` with latest `regenerator`.

**Comment by pkese** [comment]
Beats me...

If I run `make build` on a current version of babel, the `dist/polyfill.min.js` looks perfectly fine without any `.return` or `.throw`.

@sebmck I think @zloirock was right -- the make process just doesn't rebuild the `browser-polyfill`.

**Comment by sebmck** [comment]
Should be fixed as of 5.1.12.

**Comment by iam4x** [comment]
Thank's @sebmck !

**Comment by alexlafroscia** [comment]
I'm still seeing this issue with the `browser-polyfill.js` file from `babel-core` v. `5.4.7`

**Comment by sebmck** [comment]
Can you confirm this happens in latest?

On Friday, 5 June 2015, Alex LaFroscia <notifications@github.com> wrote:

> I'm still seeing this issue with the browser-polyfill.js file from
> babel-core v. 5.4.7
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/babel/babel/issues/1286#issuecomment-109381343>.
>


-- 
Sebastian McKenzie

**Comment by alexlafroscia** [comment]
@sebmck I just pulled the latest `babel-core` release and imported the `browser-polyfill.js` file.  I'm seeing an `Expected Identifier` error coming from

> line 2880 character 39

which translates to this location in the source

```javascript
// Avoid propagating enqueueResult failures to Promises returned by
// later invocations of the iterator, and call generator.return() to
// allow the generator a chance to clean up.
previousPromise = enqueueResult.catch(invokeReturn);
```

Is this a separate issue?  If I should make an issue in a different location, just let me know.

**Comment by sebmck** [comment]
It's a regression in [regenerator](https://github.com/facebook/regenerator/blob/master/runtime.js#L171).

**Comment by sebmck** [comment]
I created a PR facebook/regenerator#204.

**Comment by damassi** [comment]
@sebmck - this looks like it has been merged, how do we go about integrating it back into the codebase?

**Comment by iam4x** [comment]
@sebmck Thank's again :+1: 

@damassi If you update babel it should be good, since babel is requiring latest version of regenerator (https://github.com/babel/babel/blob/master/package.json#L66)

**Comment by damassi** [comment]
Awesome -- works great ++  We're only a few months away from no more IE 8 (and a considerable diminishing of panic every time something breaks :)

**Comment by theredpea** [comment]
Hi all, am I right that the latest version of regenerator added back [delegator.iterator.return](https://github.com/facebook/regenerator/commit/ab61c20c2814110229df555c92a4237d37650f6f#diff-b1fb3ac85a28d1b282db3d98d7252a3dR339) (vs ["return"])?

**Comment by vetoshko** [comment]
I still have the same issue ("version": "6.26.0").

**Comment by ghgugle** [comment]
Same here with version 6.26.0

**Comment by abbyfleming** [comment]
@theredpea I'm noticing this as well on v6.26.0
```if (context.method === "throw") {
        if (delegate.iterator.return) {
```

**Comment by someoneinatree** [comment]
I thought this was babel problem but remidied (partially, so far...) with `new UglifyJsPlugin({ ..., uglifyOptions: { ..., ie8: true } })`

**Comment by larsgw** [comment]
`ie8: true` in the uglify json config is not doing it for me on IE9 and babel `7.0.0-beta.49`. Either way, it also happens on unminified builds.

**Comment by andrewplummer** [comment]
I'm having this issue as well: line 7178 of the unminified script has `deletgate.iterator.return`. Additionally there is a reserved word `delete:` in the minified script so neither will load for me in IE8 now. Confirmed in `7.0.0-beta.56` and `6.26.0`.

**Comment by afreidz** [comment]
@andrewplummer same here.  seems like this has resurfaced.

**Comment by loganfsmyth** [comment]
I've posted a fix in https://github.com/facebook/regenerator/pull/355 so once that lands, the following Babel release should include the fix.

**Comment by larsgw** [comment]
This could probably help in some cases too: https://babeljs.io/docs/en/next/babel-plugin-transform-member-expression-literals (as @loganfsmyth mentioned in chat).
